src/commandargs.cpp

```diff
@@ -5,16 +5,27 @@
 
 static ProgramConfiguration* Config = nullptr;
 static Setting* CurrentState = nullptr;
+String programName;
 
 bool MatchesFlag(String token)
 {
+	if(token.length() < 1)
+	{
+		return false;
+	}
+
     for(Setting& s: *Config)
     {
         if(token == s.Flag)
         {
             CurrentState = &s;
             return true;
         }
+        else if(token.at(0) == '-')
```

Comment:
Wouldn't we want this outside of the for loop so after we've checked all the settings we report an unknown flag?

Reply:
If there's an unknown flag the program is being called in an invalid manner, so my interpretation would be to just fail first since, if there's a failed flag, the user isn't invoking the behavior of the program correctly, so proceeding will result in the program not doing what the user specified by using the incorrect call form. 

e.g. you wouldn't want a call to a function to continue if you passed the wrong arguments to it, but here we can be helpful and inform the user to try again :)

Reply:
This is subjective though, and can be changed

Reply:
I agree with your approach, but what I meant was that the `else if` is living in the for loop. If for example the settings contains the known flags `-a, -b, -c, -d` and the `token` matched `-d`, the `else if` would be executed providing usage, which may not be desirable in this case

Reply:
Ah, I see now, this makes sense, I'll pull it out of the loop in the next revision